plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -215,18 +215,28 @@ export class PgInstrumentation extends InstrumentationBase {
         // Modify query text w/ a tracing comment before invoking original for
         // tracing, but only if args[0] has one of our expected shapes.
         //
-        // TODO: remove the `as ...` casts below when the TS version is upgraded.
+        // TODO: remove the `as ...` casts on `arg0` below when the TS version is upgraded.
         // Newer TS versions will use the result of firstArgIsQueryObjectWithText
         // to properly narrow arg0, but TS 4.3.5 does not.
         if (instrumentationConfig.addSqlCommenterCommentToQueries) {
+          const customSqlCommenterAttributes = context
+            .active()
+            .getValue(
+              Symbol('customSqlCommenterAttributes')
```

Comment:
I'm not sure about this bit, any suggestions are welcomed

Reply:
This won't work. This names the symbol but it is still unique each time it is called. You would need to use `Symbol.for` in order to get the same symbol each time, but I think this approach loses some of the advantage of using a symbol.

Using context seems like a reasonable way to go about this but I haven't seen any other instrumentations do this in any language I think. I think the idea of using context to inject things into instrumentations has merit, but maybe the mechanism can be generalized to other instrumentations as well.

@open-telemetry/javascript-maintainers what would you think?

Reply:
> You would need to use Symbol.for in order to get the same symbol each time, but I think this approach loses some of the advantage of using a symbol.

You're quite right, I was forced to use a `Symbol` here because that's what the API expects. I think if we came up with a standard way of using the context to pass data to instrumentations, then there'd be a different API here. 

I'm not sure I can offer much insight into what the API could look like but I'm keen to hear what the others think and very happy to implement what is agreed upon :raised_hands: